src/Tests/Cluster/TaskManagement/TasksCancel/TasksCancelApiTests.cs

```diff
@@ -10,29 +10,31 @@
 
 namespace Tests.Cluster.TaskManagement.TasksCancel
 {
-	// TODO Unit tests will fail in mixed mode because SetupTaskIds isn't setup
 	[Collection(IntegrationContext.OwnIndex)]
 	public class TasksCancelApiTests : ApiIntegrationTestBase<ITasksCancelResponse, ITasksCancelRequest, TasksCancelDescriptor, TasksCancelRequest>
 	{
-		public class Test
+		private TaskId TaskId => this.RanIntegrationSetup ? this.ExtendedValue<TaskId>("taskId") : "foo:1";
+
+		private class Test
 		{
 			public long Id { get; set; }
 			public string Flag { get; set; }
 		}
 
 		public TasksCancelApiTests(OwnIndexCluster cluster, EndpointUsage usage) : base(cluster, usage) { }
 
-		protected IDictionary<string, TaskId> SetupTaskIds { get; } = new Dictionary<string, TaskId>();
-
-		protected override void BeforeAllCalls(IElasticClient client, IDictionary<ClientMethod, string> values)
+		protected override void IntegrationSetup(IElasticClient client, CallUniqueValues values)
```

Comment:
I like the rename.  Makes it clear that this is only applies to integration tests.